tasks/parameter/sp.yml

```diff
@@ -21,8 +21,8 @@
 
 - name: Set service principal parameter
   set_fact:
-      sp: "{{ sp | default({}) | combine({ item.key: vars[item.value] }) }}"
+      sp: "{{ sp | default({}) | combine({ item.key: item.value }) }}"
```

Comment:
We have already transformed `value` so we can assign it directly.

NB. Using `var[key]` doesn't perform a jinja transform